app/domain/authentication/authn_jwt/validate_status.rb

```diff
@@ -0,0 +1,124 @@
+
+      def call
+        @logger.info(LogMessages::Authentication::AuthnJwt::ValidatingJwtStatusConfiguration.new)
+        validate_generic_status_validations
+        validate_secrets
+        @logger.info(LogMessages::Authentication::AuthnJwt::ValidatedJwtStatusConfiguration.new)
+      end
+
+      private
+
+      def validate_generic_status_validations
+        validate_account_exists
+        validate_service_id_exists
+        validate_user_has_access_to_status_webservice
+        validate_authenticator_webservice_exists
+        validate_webservice_is_whitelisted
+      end
+
+      def validate_account_exists
+        @validate_account_exists.(
+          account: account
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedAccountExists.new)
+      end
+
+      def validate_service_id_exists
+        raise Errors::Authentication::AuthnJwt::ServiceIdMissing unless service_id
+
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedServiceIdExists.new)
+      end
+
+      def validate_user_has_access_to_status_webservice
+        @validate_role_can_access_webservice.(
+          webservice: status_webservice,
+          account: account,
+          user_id: username,
+          privilege: 'read'
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedUserHasAccessToStatusWebservice.new)
+      end
+
+      def validate_authenticator_webservice_exists
+        @validate_webservice_exists.(
+          webservice: webservice,
+          account: account
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedAuthenticatorWebServiceExists.new)
+      end
+
+      def validate_webservice_is_whitelisted
+        @validate_webservice_is_whitelisted.(
+          webservice: webservice,
+          account: account,
+          enabled_authenticators: @enabled_authenticators
+        )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedStatusWebserviceIsWhitelisted.new)
+      end
+
+      def validate_secrets
+        validate_signing_key_secrets
+        validate_issuer
+        validate_identity_secrets
+      end
+
+      def validate_signing_key_secrets
+        @create_signing_key.call(authentication_parameters: authentication_parameters)
+      end
+
+      def validate_issuer
+        @fetch_issuer_value.call(authentication_parameters: authentication_parameters)
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedIssuerConfiguration.new)
+      end
+
+      def validate_identity_secrets
+        @identity_from_decoded_token_provider_class.new(
+          authentication_parameters: authentication_parameters
+        ).identity_configured_properly?
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedIdentityConfiguration.new)
+      end
+
+      def authentication_parameters
+        @authentication_parameters ||= Authentication::AuthnJwt::AuthenticationParameters.new(
```

Comment:
It looks like the `Authentication::AuthnJwt::AuthenticationParameters` object could be passed into this class's `call` method (as it's unique to the request, not the authenticator setup). This would help with mocking and testing.